### PR TITLE
chore: remove lazy_static in favor of LazyLock

### PR DESCRIPTION
**Pull request by hntd187**
Turns out we don't use lazy static in many places so potentially removing the dependency is straight forward. Lazy static itself even gives an example from the standard library. 

https://github.com/rust-lang-nursery/lazy-static.rs?tab=readme-ov-file#standard-library

### CODE REVIEW

File: kernel/src/scan/data_skipping.rs [hunk 1]
```diff
@@ -180,13 +180,12 @@ impl DataSkippingFilter {
...
-        lazy_static::lazy_static!(
-            static ref PREDICATE_SCHEMA: DataType = StructType::new(vec![
-                StructField::new("predicate", DataType::BOOLEAN, true),
-            ]).into();
-            static ref STATS_EXPR: Expr = Expr::column("add.stats");
-            static ref FILTER_EXPR: Expr = Expr::column("predicate").distinct(Expr::literal(false));
-        );
+        static PREDICATE_SCHEMA: LazyLock<DataType> = LazyLock::new(|| {
+            StructType::new(vec![StructField::new("predicate", DataType::BOOLEAN, true)]).into()
+        });
```
Comment (scovich) [inline kernel/src/scan/data_skipping.rs:185]:
```suggestion
        static PREDICATE_SCHEMA: LazyLock<DataType> = LazyLock::new(|| {
            DataType::struct_type(vec![StructField::new("predicate", DataType::BOOLEAN, true)])
        });
```

**Review by scovich — APPROVED** [review]
Nice. Should we also remove the `lazy_static` crate, tho?

**Comment by hntd187** [comment]
A checkout reverted my removal of the crate, we should be good to go now.

**Review by nicklan — APPROVED** [review]
lgtm! thanks

**Pull request merged**